src/transformers/utils/import_utils.py

```diff
@@ -359,7 +359,7 @@ def is_torch_fp16_available_on_device(device):
     import torch
 
     try:
-        x = torch.zeros(2, 2, dtype=torch.float16).to(device)
+        x = torch.zeros(2, 2, dtype=torch.float16).to("{}:0".format(device))
```

Comment:
Ascend NPU currently does not support `torch.zeros(2, 2).to("npu")` :(
And This modification has been tested on mps and cpu device.